msdk-rawdata/msdk-rawdata-peakinvestigator/src/main/java/io/github/msdk/rawdata/peakinvestigator/PeakInvestigatorMsSpectrum.java

```diff
@@ -0,0 +1,125 @@
+/* 
+ * (C) Copyright 2015-2016 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+
+package io.github.msdk.rawdata.peakinvestigator;
+
+import java.util.TreeMap;
+
+import com.google.common.collect.Range;
+
+import io.github.msdk.datamodel.msspectra.MsSpectrum;
+import io.github.msdk.datamodel.msspectra.MsSpectrumType;
+import io.github.msdk.util.tolerances.MzTolerance;
+
+/**
+ * This class is used provide m/z-dependent mass tolerances from
+ * PeakInvestigator by decorating an existing instance of MsSpectrum.
+ *
+ */
+public class PeakInvestigatorMsSpectrum implements MsSpectrum, MzTolerance {
```

Comment:
This is fine, but keep in mind that if you add a new implementation of MsSpectrum, it has to be updated every time there is a change in the MsSpectrum interface. If all modules add their own implementations, it becomes a hassle to maintain them all.
I think you could achieve the same without adding a new implementation, simply using a container class that contains both an MsSpetrum and an MzTolerance, accessible by getters getMsSpectrum() and getMzTolerance().